spyder/app/mainwindow.py

```diff
@@ -3060,9 +3060,11 @@ def add_to_fileswitcher(self, plugin, tabs, data, icon):
             self.fileswitcher = FileSwitcher(self, plugin, tabs, data, icon)
         else:
             self.fileswitcher.add_plugin(plugin, tabs, data, icon)
-
-        self.fileswitcher.sig_goto_file.connect(
+        try:
+            self.fileswitcher.sig_goto_file.connect(
                 plugin.get_current_tab_manager().set_stack_index)
+        except:
+            pass

```

Comment:
@juanis2112, please revert this change.

Reply:
Also, as a general note, you should essentially *never* use a bare except (equivalent to ``except BaseException:``, vs, at the very least, ``except Exception:``), as the former incorrectly catch ``SystemExit``, ``KeyboardInterrupt`` and ``GeneratorExit`` and prevents Spyder from being correctly closed by these if happens to be in the block in question (or gets stuck in it, etc.). 

Furthermore, if you do ``except Exception:``, you should think very carefully about what purpose it serves and include a comment justifying it over something more specific and tailored, as it will also hide errors from our tests and error reporting system, silently retaining potentially fixable problems.

Reply:
@juanis2112, what I meant by "revert" is that you need to leave things as they were before. This means removing the try/except you added here, so this change doesn't appear as part of this PR.